src/Hooks.ts

```diff
@@ -143,3 +145,75 @@ export function useFocusState() {
 
   return focusState;
 }
+
+type EffectCallback = (() => void) | (() => () => void);
+
+// Inspired by same hook from react-navigation v5
+// See https://github.com/react-navigation/hooks/issues/39#issuecomment-534694135
+export const useFocusEffect = (callback: EffectCallback) => {
+  const navigation = useNavigation();
+  const getCallback = useGetter(callback);
+
+  useEffect(() => {
+    let isFocused = false;
+    let cleanup: (() => void) | void;
+
+    if (navigation.isFocused()) {
+      cleanup = getCallback()();
+      isFocused = true;
+    }
+
+    const focusSubscription = navigation.addListener('willFocus', () => {
+      // If callback was already called for focus, avoid calling it again
+      // The focus event may also fire on intial render, so we guard against runing the effect twice
+      if (isFocused) {
+        return;
+      }
+
+      cleanup && cleanup();
+      cleanup = getCallback()();
+      isFocused = true;
+    });
+
+    const blurSubscription = navigation.addListener('willBlur', () => {
+      cleanup && cleanup();
+      cleanup = undefined;
+      isFocused = false;
+    });
+
+    return () => {
+      cleanup && cleanup();
+      focusSubscription.remove();
+      blurSubscription.remove();
+    };
+  }, [getCallback, navigation]);
+};
+
+export const useIsFocused = () => {
+  const navigation = useNavigation();
+  const [focused, setFocused] = useState(navigation.isFocused());
```

Comment:
Maybe:

```suggestion
  const [focused, setFocused] = useState(navigation.isFocused);
```

Since we don't need to call `isFocused` every-render. It's not expensive afaik, but can be skipped.

Reply:
ahhh yeah, I thought in v5 isFocused was a mutable variable on navigation by reading your code, didn't think of useState actually executing the function :D